src/registry/textfield/validationSchema.ts

```diff
@@ -7,18 +7,27 @@ const getValidationSchema: GetValidationSchema<TextFieldComponentSchema> = (
   componentDefinition,
   {validatePlugins}
 ) => {
-  const {key, validate = {}, multiple} = componentDefinition;
+  const {key, validate = {}, multiple, errors} = componentDefinition;
   const {required, maxLength, pattern, plugins = []} = validate;
 
-  let schema: z.ZodFirstPartySchemaTypes = z.string();
-  if (maxLength !== undefined) schema = schema.max(maxLength);
+  let schema: z.ZodFirstPartySchemaTypes = z.string(
+    errors?.required ? {required_error: errors.required} : undefined
+  );
```

Comment:
```suggestion
  let schema: z.ZodFirstPartySchemaTypes = z.string({required_error: errors?.required});
```
tests still pass if you do it this way and apply this pattern everywhere - the types allow for `required_error` / `message` itself to be undefined, so we can use that to short-circuit and avoid duplication.